index.d.ts

```diff
@@ -39,11 +39,10 @@ declare module 'i18next' {
 type ObjectOrNever = TypeOptions['allowObjectInHTMLChildren'] extends true
   ? Record<string, unknown>
   : never;
-type ReactI18NextChild = React.ReactNode | ObjectOrNever;
 
 declare module 'react' {
   interface HTMLAttributes<T> {
-    children?: ReactI18NextChild | Iterable<ReactI18NextChild>;
+    children?: React.ReactNode | ObjectOrNever;
```

Comment:
Now, children will behave exactly like the [actual](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/v17/index.d.ts#L1342) unless `allowObjectInHTMLChildren` is `true`.